src/psyclone/generator.py

```diff
@@ -187,6 +194,10 @@ def generate(filename, api="", kernel_path="", script_name=None,
                 "generate: Unsupported API '{0}' specified. Supported "
                 "types are {1}.".format(api, Config.get().supported_apis))
 
+    # Store Kernel-output options in our Configuration object
+    Config.get().kernel_output_dir = kern_out_path
+    Config.get().kernel_naming = kern_naming
```

Comment:
This could theoretically raise an exception even though you limit the valid values. Should you check?

Reply:
I should and now I do :-)